libqtile/backend/base.py

```diff
@@ -93,6 +96,13 @@ def flush(self) -> None:
     def graceful_shutdown(self):
         """Try to close windows gracefully before exiting"""
 
+    def simulate_keypress(self, modifiers: list[str], key: str) -> None:
```

Comment:
I think this will need to use `typing.List`, using the built-in type is supported in [Python 3.9](https://www.python.org/dev/peps/pep-0585/), but is not supported before that. Similarly for some of the other `[]` types annotations.

Reply:
Ah yes. This was actually an intentional decision, though I neglected to bring it up for discussion in the post. As you said, built-in types are support for 3.9+ only. The rationale behind making this change was that the CI for mypy uses 3.9, so we could start to phase out the legacy types.

Then again, this might become problematic if people are trying to run mypy locally with older versions of Python. I guess that is the case?

Reply:
I've updated these to use the `typing` versions for maintain compability with older versions.